File: app/src/main/java/ua/hospes/rtm/core/ui/RxScrollListener.java
package ua.hospes.rtm.core.ui;

import android.support.v7.widget.RecyclerView;

import io.reactivex.subjects.PublishSubject;

/**
 * @author Andrew Khloponin
 */
public abstract class RxScrollListener extends RecyclerView.OnScrollListener {
    protected final PublishSubject<RecyclerView> subject = PublishSubject.create();


    @Override
    public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
        subject.onNext(recyclerView);
    }

    @Override
    public void onScrollStateChanged(RecyclerView recyclerView, int newState) {
        if (newState == RecyclerView.SCROLL_STATE_IDLE) {
            subject.onNext(recyclerView);
        }
    }


    public void forceUpdate(RecyclerView recyclerView) {
        subject.onNext(recyclerView);
    }
}

File: app/src/main/java/ua/hospes/rtm/core/ui/RxScrollListener.kt
package ua.hospes.rtm.core.ui

import androidx.recyclerview.widget.RecyclerView

import io.reactivex.subjects.PublishSubject

/**
 * @author Andrew Khloponin
 */
abstract class RxScrollListener : RecyclerView.OnScrollListener() {
    protected val subject = PublishSubject.create<RecyclerView>()


    override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
        subject.onNext(recyclerView)
    }

    override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {
        if (newState == RecyclerView.SCROLL_STATE_IDLE) {
            subject.onNext(recyclerView)
        }
    }


    fun forceUpdate(recyclerView: RecyclerView) {
        subject.onNext(recyclerView)
    }
}